### PR TITLE
ISPN-12801 Hot Rod client logs a timeout error after every server error

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.redhat.com/browse/ISPN-12801

* Cancel the timeout future on completeExceptionally().
* Add a test for cache creation failure.
* Improve logging for admin and execute operations.

### CODE REVIEW

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/HotRodOperation.java [hunk 1]
```diff
@@ -39,6 +39,7 @@
 @Immutable
 public abstract class HotRodOperation<T> extends CompletableFuture<T> implements HotRodConstants, Runnable {
    private static final Log log = LogFactory.getLog(HotRodOperation.class);
+   private static final boolean trace = log.isTraceEnabled();
```
Comment (wburns) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/HotRodOperation.java:42]: This appears to be unused?

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/admin/RemoteCacheAdminTest.java [hunk 2]
```diff
@@ -96,9 +107,25 @@ public void cacheCreateRemoveTest(Method m) {
       assertFalse(manager(1).cacheExists(cacheName));
    }
 
+   public void cacheCreateMissingTemplate(Method m) throws InterruptedException {
+      Exceptions.expectException(HotRodClientException.class,
+                                 () -> client(0).administration().withFlags(CacheContainerAdmin.AdminFlag.VOLATILE)
+                                                .createCache("some_name", "some_template_name"));
+      Thread.sleep(999);
```
Comment (wburns) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/admin/RemoteCacheAdminTest.java:114]: Is this sleep needed?

Comment (wburns) [reply client/hotrod-client/src/test/java/org/infinispan/client/hotrod/admin/RemoteCacheAdminTest.java:114]: If so maybe we change the socket timeout to be lower and lower this as well?

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/ExecuteOperation.java [hunk 3]
```diff
@@ -68,4 +70,20 @@ protected void executeOperation(Channel channel) {
    public void acceptResponse(ByteBuf buf, short status, HeaderDecoder decoder) {
       complete(bytes2obj(channelFactory.getMarshaller(), ByteBufUtil.readArray(buf), dataFormat.isObjectStorage(), cfg.getClassAllowList()));
    }
+
+   @Override
+   protected void addParams(StringBuilder sb) {
```
Comment (wburns) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/ExecuteOperation.java:75]: This seems a bit unrelated, but fine

**Review by wburns — COMMENTED** [review]
Changes look good, however it is conflicted now as I ended up fixing the cancelling of the timeout in another PR as it causes a temporary memory leak.

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/HotRodOperation.java (hunk 1 above)
Comment (danberindei) [reply client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/HotRodOperation.java]: Yeah, I added a temporary trace log while testing and forgot to remove the field

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/ExecuteOperation.java (hunk 3 above)
Comment (danberindei) [reply client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/ExecuteOperation.java:75]: Without this change, cache creation operations are logged as `AdminOperation (default)`, with no mention of the cache name.

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/admin/RemoteCacheAdminTest.java (hunk 2 above)
Comment (danberindei) [reply client/hotrod-client/src/test/java/org/infinispan/client/hotrod/admin/RemoteCacheAdminTest.java]: Updated to remove the sleep and check the task queue in the scheduled executor instead @wburns

**Comment by wburns** [comment]
Integrated into master, thanks @danberindei !

**Pull request merged**